### PR TITLE
WIP: replace etcd 2 client code with etcd 3 client code

### PR DESCRIPTION
**Pull request by dseevr**
Please don't bother reviewing yet.  I'm only opening this so I can run the unit/system tests on our CI

More changes to come

### CODE REVIEW

**Comment by dseevr** [comment]
build pr

**Comment by dseevr** [comment]
build pr

**Comment by dseevr** [comment]
build pr

**Comment by wangxf1987** [comment]
Interesting work, and the objdb is mainly responsible for doing.

**Comment by g1rana** [comment]
lgtm

File: objdb/etcdClient.go [hunk 1]
```diff
@@ -165,23 +151,19 @@ func (ep *EtcdClient) ListDir(key string) ([]string, error) {
...
 	var retList []string
-	// Call a recursive function to recurse thru each directory and get all files
-	// Warning: assumes directory itep is not interesting to the caller
-	// Warning2: there is also an assumption that keynames are not required
-	//           Which means, caller has to derive the key from value :(
-	retList = recursAddNode(resp.Node, retList)
+
+	// convert all the keys into strings (etcd3 doesn't have directories)
+	for _, kv := range resp.Kvs {
+		retList = append(retList, string(kv.Key))
```
Comment (wangxf1987) [inline objdb/etcdClient.go]: It's should be get Value from kv, not Key. Like this:
retList = append(retList, string(kv.Value)).
In the mvccpb, It's define to the struct, 
        `type KeyValue struct {
	// key is the key in bytes. An empty key is not allowed.
	Key []byte `protobuf:"bytes,1,opt,name=key,proto3" json:"key,omitempty"`
	// create_revision is the revision of last creation on this key.
	CreateRevision int64 `protobuf:"varint,2,opt,name=create_revision,json=createRevision,proto3" json:"create_revision,omitempty"`
	// mod_revision is the revision of last modification on this key.
	ModRevision int64 `protobuf:"varint,3,opt,name=mod_revision,json=modRevision,proto3" json:"mod_revision,omitempty"`
	// version is the version of the key. A deletion resets
	// the version to zero and any modification of the key
	// increases its version.
	Version int64 `protobuf:"varint,4,opt,name=version,proto3" json:"version,omitempty"`
	// value is the value held by the key, in bytes.
	Value []byte `protobuf:"bytes,5,opt,name=value,proto3" json:"value,omitempty"`
	// lease is the ID of the lease that attached to key.
	// When the attached lease expires, the key will be deleted.
	// If lease is 0, then no lease is attached to the key.
	Lease int64 `protobuf:"varint,6,opt,name=lease,proto3" json:"lease,omitempty"`
}`

File: Vagrantfile [hunk 2]
```diff
@@ -165,9 +165,12 @@ SCRIPT
 provision_node = <<SCRIPT
 set -x
 
+curl -o /tmp/etcd.tar.gz -L https://github.com/coreos/etcd/releases/download/v3.2.10/etcd-v3.2.10-linux-amd64.tar.gz
+tar -C /tmp/ -xzf /tmp/etcd.tar.gz
+
 ## start etcd with generated config
 echo "#!/bin/bash" > /usr/bin/etcd.sh
-echo "etcd --name %{node_name} --data-dir /var/lib/etcd \
+echo "/tmp/etcd-v3.2.10-linux-amd64/etcd --name %{node_name} --data-dir /var/lib/etcd \
```
[the remark this answers is not in the document]
Comment (dseevr) [reply Vagrantfile]: This was just a temporary hack to enable development against etcd3

**Comment by dseevr** [comment]
This PR is unfinished and will never be finished since I don't work on this project anymore.  If someone wants to use this code as a base for adding etcd3 support, please feel free to take it

**Pull request closed without merging**